packages/core/src/messagingProtocols/proteus/ProteusService/ProteusService.ts

```diff
@@ -256,51 +243,30 @@ export class ProteusService {
     }
   }
 
+  public deleteSession(userId: QualifiedId, clientId: string) {
+    return deleteSession({
+      userId,
+      clientId,
+      cryptoClient: this.cryptoClient,
+    });
+  }
+
   public async encrypt(
     plainText: Uint8Array,
-    recipients: UserPreKeyBundleMap | UserClients,
-    domain: string = '',
+    recipients: QualifiedUserPreKeyBundleMap | QualifiedUserClients,
   ): Promise<EncryptionResult> {
     const {sessions, unknowns} = await initSessions({
       recipients,
-      domain,
       apiClient: this.apiClient,
       cryptoClient: this.cryptoClient,
       logger: this.logger,
     });
 
-    const payload = await this.cryptoClient.encrypt(sessions, plainText);
-
-    return {payloads: buildEncryptedPayloads(payload), unknowns};
-  }
-
-  public deleteSession(userId: QualifiedId, clientId: string) {
-    return deleteSession({
-      userId,
-      clientId,
-      useQualifiedIds: this.config.useQualifiedIds,
-      cryptoClient: this.cryptoClient,
-    });
-  }
-
-  public async encryptQualified(
-    plainText: Uint8Array,
-    preKeyBundles: QualifiedUserPreKeyBundleMap | QualifiedUserClients,
-  ): Promise<{payloads: QualifiedOTRRecipients; unknowns?: QualifiedUserClients}> {
-    const qualifiedOTRRecipients: QualifiedOTRRecipients = {};
-    const missingRecipients: QualifiedUserClients = {};
-
-    for (const [domain, preKeyBundleMap] of Object.entries(preKeyBundles)) {
-      const {unknowns, payloads} = await this.encrypt(plainText, preKeyBundleMap, domain);
-      qualifiedOTRRecipients[domain] = payloads;
-      if (unknowns) {
-        missingRecipients[domain] = unknowns;
-      }
-    }
+    const payloads = await this.cryptoClient.encrypt(sessions, plainText);
```

Comment:
encryption becomes quite trivial now that we don't need to account for non-qualified session. We can just bulk encrypt all the payloads at once (and avoid a few back and forth between JS and wasm lands)